src/package.coffee

```diff
@@ -26,16 +26,11 @@ class Package
     packagePath?.startsWith(@resourcePathWithTrailingSlash)
 
   @normalizeMetadata: (metadata) ->
-    if typeof metadata.repository is 'string'
-      metadata.repository =
-        type: 'git'
-        url: metadata.repository
-
-    repoUrl = metadata.repository?.url
-    if repoUrl
-      info = hostedGitInfo.fromUrl(repoUrl)
-      if info?.getDefaultRepresentation() is 'shortcut'
-        metadata.repository.url = info.https().replace(/^git\+/, '')
+    unless metadata?._id
+      normalizePackageData ?= require 'normalize-package-data'
+      normalizePackageData(metadata)
+      if metadata.repository?.type is 'git' and typeof metadata.repository.url is 'string'
```

Comment:
@jlord after reviewing this again, I thought it would be better to tighten up this check to only call `replace` if the url is  a `string`.

That way in the weird case where they have a number of boolean value there, it won't blow up with a `undefined is not a function` errors.